cmd/shipperctl/configurator/cluster.go

```diff
@@ -0,0 +1,312 @@
+			Name:     clusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      subjectName,
+				Namespace: subjectNamespace,
+			},
+		},
+	}
+
+	_, err := c.KubeClient.RbacV1().ClusterRoleBindings().Create(clusterRoleBinding)
+
+	return err
+}
+
+func (c *Cluster) CreateRole(name, namespace string) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1.PolicyRule{
+				APIGroups: []string{"v1"},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"GET"},
+			},
+		},
+	}
+
+	_, err := c.KubeClient.RbacV1().Roles(namespace).Create(role)
+
+	return err
+}
+
+func (c *Cluster) CreateRoleBinding(roleBindingName, roleName, subjectName, subjectNamespace string) error {
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: subjectNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     roleName,
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      subjectName,
+				Namespace: subjectNamespace,
+			},
+		},
+	}
+
+	_, err := c.KubeClient.RbacV1().RoleBindings(subjectNamespace).Create(roleBinding)
+
+	return err
+}
+
+func (c *Cluster) ShouldCopySecret(name, namespace string) (bool, error) {
+	_, err := c.KubeClient.CoreV1().Secrets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return true, nil
+		} else {
+			return false, err
+		}
+	}
+
+	return false, nil
+}
+
+func (c *Cluster) FetchSecretForServiceAccount(name, namespace string) (*corev1.Secret, error) {
+	serviceAccount, err := c.KubeClient.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(serviceAccount.Secrets) == 0 {
+		return nil, NewSecretNotPopulatedError(serviceAccount)
+	}
+
+	secretName := serviceAccount.Secrets[0].Name
+	secret, err := c.KubeClient.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+func (c *Cluster) FetchCluster(clusterName string) (*shipperv1.Cluster, error) {
+	return c.ShipperClient.ShipperV1alpha1().Clusters().Get(clusterName, metav1.GetOptions{})
+}
+
+func (c *Cluster) CopySecret(cluster *shipperv1.Cluster, newNamespace string, secret *corev1.Secret) error {
+	hash := crc32.NewIEEE()
+	hash.Write(secret.Data["ca.crt"])
+	hash.Write(secret.Data["token"])
+
+	newSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cluster.Name,
+			Namespace: newNamespace,
+			Annotations: map[string]string{
+				shipperv1.SecretChecksumAnnotation: hex.EncodeToString(hash.Sum(nil)),
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				metav1.OwnerReference{
+					APIVersion: "shipper.booking.com/v1",
```

Comment:
Does it make any sense to keep it in a constant?

Reply:
Particularly given that it's the wrong version! This got me thinking where one might find a constant with this available, and as it turns out, I don't think there is such a thing anywhere. I guess we can drop one into apis/v1alpha somewhere.

I think it'd be ok to just change the string to be 'v1alpha1' in this PR, and let's try to fix all the version strings in another round; there are many more than this one IIRC.

Reply:
Okay, will keep this as is, so that it gets fixed after merging pull request #49.